app/scripts/models/auth_brokers/oauth-redirect.js

```diff
@@ -68,7 +69,19 @@ define(function (require, exports, module) {
           if (result.action) {
             extraParams['action'] = result.action;
           }
-          win.location.href = Url.updateSearchString(result.redirect, extraParams);
```

Comment:
I'm surprised we did this `win.location.href =` here, we should have been delegating to the router, the router would have flushed the metrics and then redirected.